script.py

```diff
@@ -25,45 +25,45 @@ def generate_token(client_id, client_secret):
 
     if(token_response.status_code == 200):
         return token_response.json()['access_token']
-    else:
-        print('Invalid credentials')
 
 
-def request_vehicles(*arg):
-    if len(arg) == 1:
-        return requests.get(f'{osu_API}v1/beaverbus/vehicles', headers={
-            'Authorization': f'Bearer {arg[0]}'
-        })
-    if len(arg) == 2:
-        return requests.get(f'{osu_API}v1/beaverbus/vehicles/{arg[1]}',
-                            headers={'Authorization': f'Bearer {arg[0]}'}
+def request_vehicles(id):
```

Comment:
You can give the `id` a default value like `request_vehicles(id=None)` so that you won't need to always call this function with `None` value if you wanna search with all vehicles. You only need to pass an id to overwrite the default value when needed.